### PR TITLE
Import checks from `applications-superstaq`

### PR DESCRIPTION
**Pull request by perlinm**
this should work after [`app-ss#70`](https://github.com/SupertechLabs/applications-superstaq/pull/70)

todo in a future PR: add notebook check to workflows

### CODE REVIEW

File: dev-requirements.txt [hunk 1]
```diff
@@ -1,8 +1,10 @@
 black[jupyter]~=22.3.0
-flake8==3.8.4
 flake8-import-order==0.18.1
-mypy==0.910
-pylint==2.6.0
-pytest==6.1.2
+flake8==3.8.4
```
Comment (vtomole) [inline dev-requirements.txt:3]:
```suggestion
flake8~=3.8.4
```

File: dev-requirements.txt [hunk 2]
```diff
@@ -1,8 +1,10 @@
...
+mypy==0.961
```
Comment (vtomole) [inline dev-requirements.txt:4]:
```suggestion
mypy~=0.961
```

**Review by vtomole — COMMENTED** [review]
Some nits then LGTM

File: dev-requirements.txt (hunk 1 above)
Comment (perlinm) [reply dev-requirements.txt]: done

**Comment by perlinm** [comment]
Checks pass for me locally but on the server I now get (from `mypy`):
`cirq_superstaq/custom_gates.py:426: error: Name "cirq.ion.ion_gates.MSGate" is not defined  [name-defined]`
and similar errors from `pytest/coverage`, such as
```
cirq_superstaq/custom_gates.py:426: in <module>
    class MSGate(cirq.ion.ion_gates.MSGate):
E   AttributeError: module 'cirq.ion' has no attribute 'ion_gates'
```

This started happening after loosening requirements for `flake8` and `mypy` ([commit diff](https://github.com/SupertechLabs/cirq-superstaq/pull/275/commits/55484406316169859d0375df09321f4ee343edd8)), so it seems like there's an unrelated/external issue causing it(?)

**Comment by perlinm** [comment]
@vtomole ready for re-review.  Note: dev-requirements are about to change in #286

**Comment by perlinm** [comment]
Github stopped running checks for some reason?

File: dev-requirements.txt [hunk 3]
```diff
@@ -1,8 +1,10 @@
 black[jupyter]~=22.3.0
-flake8==3.8.4
 flake8-import-order==0.18.1
-mypy==0.910
-pylint==2.6.0
-pytest==6.1.2
+flake8~=3.8.4
+mypy~=0.961
```
Comment (richrines1) [inline dev-requirements.txt:4]: nit
```suggestion
mypy>=0.961
```
(they're synonymous in this case but imo `>=` is more clear)

File: dev-requirements.txt [hunk 4]
```diff
@@ -1,8 +1,10 @@
 black[jupyter]~=22.3.0
-flake8==3.8.4
 flake8-import-order==0.18.1
-mypy==0.910
-pylint==2.6.0
-pytest==6.1.2
+flake8~=3.8.4
```
Comment (richrines1) [inline dev-requirements.txt:3]: wdyt about making these all `>=` instead of `~=`?

**Comment by perlinm** [comment]
@richrines1 these will become obsolete after [`css#286`](https://github.com/SupertechLabs/cirq-superstaq/pull/286).  Would be okay with changing dev-requirements in `app-ss` though.

**Comment by richrines1** [comment]
> @richrines1 these will become obsolete after [`css#286`](https://github.com/SupertechLabs/cirq-superstaq/pull/286). Would be okay with changing dev-requirements in `app-ss` though.

ahh got it

**Comment by richrines1** [comment]
out of curiosity would this PR be any more complex if it was merged with #286? it seems like a lot of the changes in that PR overlap a temporary change here, so it might simplify the history a bit to combine them...

**Comment by perlinm** [comment]
Sure, I can do that.  Closing.

**Pull request closed without merging**